### PR TITLE
mds: add uptime to MDS status

### PR DESCRIPTION
**Pull request by batrick**
Fixes: https://tracker.ceph.com/issues/23094

Signed-off-by: Patrick Donnelly <pdonnell@redhat.com>

### CODE REVIEW

File: src/mds/MDSDaemon.h [hunk 1]
```diff
@@ -47,6 +47,15 @@ class MDSDaemon : public Dispatcher, public md_config_obs_t {
 
   SafeTimer    timer;
 
+
+  mono_time get_starttime() const {
+    return starttime;
+  }
```
Comment (joscollin) [inline src/mds/MDSDaemon.h]: Unused function? Or am I missing something?

File: src/mds/MDSRank.h [hunk 2]
```diff
@@ -138,6 +138,14 @@ class MDSRank {
     // a separate lock here in future potentially.
     Mutex &mds_lock;
 
+    mono_time get_starttime() const {
+      return starttime;
+    }
```
Comment (joscollin) [inline src/mds/MDSRank.h]: ditto

File: src/mds/MDSRank.h [hunk 3]
```diff
@@ -138,6 +138,14 @@ class MDSRank {
...
+    chrono::duration<double> get_uptime() const {
+      mono_time now = mono_clock::now();
+      return chrono::duration<double>(now-starttime);
+    }
```
Comment (joscollin) [inline src/mds/MDSRank.h]: ditto

File: src/mds/MDSDaemon.h [hunk 4]
```diff
@@ -47,6 +47,15 @@ class MDSDaemon : public Dispatcher, public md_config_obs_t {
...
+  chrono::duration<double> get_uptime() const {
+    mono_time now = mono_clock::now();
+    return chrono::duration<double>(now-starttime);
+  }
+
```
Comment (joscollin) [inline src/mds/MDSDaemon.h]: Can be `private`.

File: src/mds/MDSDaemon.h (hunk 1 above)
Comment (batrick) [reply src/mds/MDSDaemon.h]: Yes. I'm okay with having unused getters.

File: src/mds/MDSRank.h (hunk 3 above)
Comment (batrick) [reply src/mds/MDSRank.h]: this is not unused.

File: src/mds/MDSDaemon.h (hunk 4 above)
Comment (batrick) [reply src/mds/MDSDaemon.h]: getters like this can be public IMO.

**Pull request merged**